File: 5_permutations.py
def permutations(n: int) -> None:
    if n == 1:
        print(1, end=" ")
    elif n < 4:
        print("NO SOLUTION", end=" ")
    else:
        even = [i for i in range(1, n + 1) if i % 2 == 0]
        odd = [i for i in range(1, n + 1) if i % 2 == 1]
        final = even + odd
        for i in final:
            print(i, end=" ")


if __name__ == "__main__":
    n = int(input())
    permutations(n)

File: 5_permutations.cpp
#include <bits/stdc++.h>
using namespace std;

void permutations(int &n)
{
    if (n == 1)
    {
        cout << 1;
    }
    else if (n < 4)
    {
        cout << "NO SOLUTION";
    }
    else
    {
        int i;

        i = 2;
        while (i <= n)
        {
            cout << i << " ";
            i += 2;
        }

        i = 1;
        while (i <= n)
        {
            cout << i << " ";
            i += 2;
        }
    }
}

int main()
{
    int n;
    cin >> n;
    permutations(n);
}